### PR TITLE
poller: implement blist events

### PR DESCRIPTION
**Pull request by aviau**
Mes chats se sont automagiquement déplacés dans "Conferences" :tada: 

Screenshot from 2020-07-27 10-55-10

### CODE REVIEW

**Comment by aviau** [comment]
@isra17 review plz!

File: src/icq/poller.rs [hunk 1]
```diff
@@ -99,12 +99,30 @@ pub async fn process_events(
...
         for buddy in &group.buddies {
             match &buddy.user_type {
                 events::UserType::Chat => {
-                    // Do we already have a chat?
+                    let chat_info = ChatInfo {
+                        sn: buddy.aim_id.clone(),
+                        stamp: None,
+                        title: match &buddy.friendly {
+                            Some(friendly) => friendly.clone(),
+                            None => buddy.aim_id.clone(),
```
Comment (leameow) [inline src/icq/poller.rs:116]: Could `display_id` be used here?

File: src/lib.rs [hunk 2]
```diff
@@ -416,21 +416,37 @@ impl PurpleICQ {
...
+                // Should we replace the blist group?
+                if let Some(info_group) = &info.group {
+                    let should_replace_group = {
+                        match chat.get_group() {
+                            Some(mut chat_group) => !chat_group.get_name().eq(info_group),
+                            None => true,
+                        }
+                    };
+                    if should_replace_group {
+                        chat.add_to_blist(&mut self.get_or_create_group(Some(&info_group)), None);
```
Comment (leameow) [inline src/lib.rs:433]: I guess this won't duplicate the existing chat and move it instead?

Comment (leameow) [reply src/lib.rs:433]: Checked the code, should be good!

File: src/icq/poller.rs (hunk 1 above)
Comment (aviau) [reply src/icq/poller.rs:116]: AFAIK it never happens that we use aim_id. Friendly is always present in this case.

Also in the data that I have captured display_id is always the same as aim_id.

File: src/lib.rs (hunk 2 above)
Comment (aviau) [reply src/lib.rs:433]: Yep! I have read the implementation.

**Comment by leameow** [comment]
lgtm!

**Pull request merged**